schemas/outcomes-metrics.v1.schema.json

```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "metrics_outcome",
+  "$ref": "#/definitions/MetricsOutcome",
+  "definitions": {
+    "MetricsOutcome": {
+      "type": "object",
+      "properties": {
+        "timestamp": {
+          "description": "The UNIX timestamp in seconds when the outcome was created, optionally truncated to the last hour.",
+          "$ref": "#/definitions/U64"
+        },
+        "org_id": {
+          "description": "The organization for which this outcome is being sent.",
+          "$ref": "#/definitions/U64"
+        },
+        "project_id": {
+          "description": "The project for which this outcome is being sent.",
+          "$ref": "#/definitions/U64"
+        },
+        "mri": {
+          "description": "Valid metric resource identifier `<type>:<namespace>/<name>[@<unit>]` for which the outcome is being sent.",
+          "type": "string"
+        },
+        "outcome": {
+          "description": "Outcome ID, metric outcomes share the same numeric outcome ID with regular outcomes.",
+          "$ref": "#/definitions/U64"
+        },
+        "reason": {
+          "description": "Optional machine readable, free-form reason code.",
+          "type": ["string", "null"]
+        },
+        "quantity": {
+          "description": "Amount of metric buckets accepted by Relay (volume).",
+          "$ref": "#/definitions/U64"
+        },
+        "cardinality": {
+          "description": "Maximum observed cardinality of the metric.",
+          "$ref": "#/definitions/U64"
+        }
```

Comment:
I'd assume for "max cardinality" we mean the number of distinct combinations of tags keys:values observed in the bucket ? OR over a period of time ? 
This would be useful information in the schema for who decides to consume from this topic

Reply:
It is over time, current plan for custom metrics is over the span of an hour (because that's what people are interested and and what Relay will be enforcing).

I'll try to improve the description, but I was a bit vague on purpose because the time frame may change and may be different per usecase.

Reply:
> I'll try to improve the description, but I was a bit vague on purpose because the time frame may change and may be different per usecase.

I think this may highlight a concern. If the time frame changes, that likely concerns both the producer and the consumers which can easily end up in incidents where the billing code miscounts metrics for assuming a wrong time frame. max_cardinality over a day vs over a minute is likely going to mean different things for the billing code. 

I would suggest an alternative approach to make this more robust: add a `time_frame` field in the message that indicates what the time frame is. 
This puts the responsibility to manage the time frame on the consumer in an explicit way and it allows you to change the time frame in a safe and backwards compatible way as the consumer immediately knows how to interpret the message

Reply:
Introduced `cardinality_window` which is required when a `cardinality` is emitted.